### DISCUSSION TITLE
Multiple projects Tailwind upgrade how to upgrade？

### DISCUSSION THREAD

**Discussion by mark-willson** [tailwindlabs/tailwindcss#3938] [Help]
First of all, thanks to Tailwind, for saving a lot of time for our development.

We are now experiencing a problem with Tailwind upgrade, and we need to ask you for advice:

We have several projects that use Tailwind,. Currently, all of them need to upgrade to 1.9.6. The current version is as follows:

0.7.3,  
0.7.4,
1.2.0,
1.4.6,
1.8.10,
1.9.6,

Can the Tailwind, in these projects be upgraded to 1.9.6 and be perfectly compatible with pages written in the old Tailwind?

If so, how should we upgrade? If not, how should we deal with it?

If a version upgrade is complicated, can we make it possible for a project to have multiple Tailwind, new pages with new Tailwind, and old pages with old Tailwind??

In addition, the reason we can't upgrade to 2. 0 is that we currently need to be compatible with IE11 browsers.

**Comment by simonswiss** [comment]
Hey!

We take backwards compatibility (and semantic versioning) pretty seriously, and upgrading all your 1.x projects to v1.9.6 should be very straightforward.

If you're upgrading from v1.2, you'll be able to move your Purge configuration into the Tailwind config file directly, instead of in your PostCSS config (feature implemented in v1.4):

https://tailwindcss.com/docs/optimizing-for-production#basic-usage

Upgrading from v.0.x to v1.x is a little more complicated - you can find an upgrade guide here on the Tailwind CSS v1 documentation website:

https://v1.tailwindcss.com/docs/upgrading-to-v1#upgrade-steps-for-all-users

Hope it helps! 👍
